xds/internal/client/xds.go

```diff
@@ -196,18 +213,20 @@ func processHTTPFilters(filters []*v3httppb.HttpFilter, server bool) ([]HTTPFilt
 		}
 		seenNames[name] = true
 
-		httpFilter, config, err := validateHTTPFilterConfig(filter.GetTypedConfig(), true)
+		httpFilter, config, err := validateHTTPFilterConfig(filter.GetTypedConfig(), true, filter.GetIsOptional())
 		if err != nil {
 			return nil, err
 		}
+		if httpFilter == nil {
+			// Optional configs are ignored.
+			continue
+		}
 		if server {
 			if _, ok := httpFilter.(httpfilter.ServerInterceptorBuilder); !ok {
-				return nil, fmt.Errorf("httpFilter %q not supported server-side", name)
-			}
-		} else {
-			if _, ok := httpFilter.(httpfilter.ClientInterceptorBuilder); !ok {
-				return nil, fmt.Errorf("httpFilter %q not supported client-side", name)
+				return nil, fmt.Errorf("HTTP filter %q not supported server-side", name)
 			}
+		} else if _, ok := httpFilter.(httpfilter.ClientInterceptorBuilder); !ok {
```

Comment:
What if
- this is client side
- the filter is installed, but only for server
- the `optional` field is set to true

This could be that we expand a filter from server only to client side.
Do we still want to error?

Reply:
No, you're right.  Good catch.  Apparently I missed this from my testing; it was correct before I moved it out of the other function.  Fixed & added test.